File: 1_arrays_and_hashing/valid_anagram.cpp
class Solution {
  public:
	bool isAnagram(string s, string t) {
		array<int, 26> a {};
		for (const char c : s)
			a.at(c - 'a')++;
		for (const char c : t)
			a.at(c - 'a')--;
		for (auto i : a)
			if (i)
				return false;

		return true;
	}
};

File: 01_arrays_and_hashing/valid_anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """

        if len(set(s)) != len(set(t)):
            return False
        else:
            for char in set(s):
                if (s.count(char) != t.count(char)):
                    return False
            return True
